gemini/rag-engine/rag_engine_evaluation.ipynb

```diff
@@ -0,0 +1,983 @@
+      "metadata": {
+        "id": "qDCv3clrN616"
+      },
+      "outputs": [],
+      "source": [
+        "# Specify your rag corpus ID here that you want to use.\n",
+        "RAG_CORPUS_ID = \"\"  # @param {type: \"string\"}\n",
+        "\n",
+        "rag_corpus = rag.get_corpus(\n",
+        "    name=f\"projects/{PROJECT_ID}/locations/{LOCATION}/ragCorpora/{RAG_CORPUS_ID}\"\n",
+        ")\n",
+        "\n",
+        "print(rag_corpus)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "DXsmEoZNOPLo"
+      },
+      "source": [
+        "# 2. Run Retrieval Quality Evaluation\n",
+        "\n",
+        "For Retrieval Quality Evaluation, we focus on the following metrics:\n",
+        "\n",
+        "- **Recall@k:**\n",
+        "  - Measures how many of the relevant documents/chunks are successfully retrieved within the top k results\n",
+        "  - Helps evaluate the retrieval component's ability to find ALL relevant information\n",
+        "- **Precision@k:**\n",
+        "  - Measures the proportion of retrieved documents that are actually relevant within the top k results\n",
+        "  - Helps evaluate how \"focused\" your retrieval is\n",
+        "- **nDCG@K:**\n",
+        "  - Measures both relevance AND ranking quality\n",
+        "  - Takes into account the position of relevant documents\n",
+        "\n",
+        "Follow the Notebook to get these metrics numbers for your configurations, and to optimize your settings."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "m5sz02OdOPYk"
+      },
+      "source": [
+        "### Define evaluation helper function."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 38,
+      "metadata": {
+        "id": "VUHxSZ7hwZ6k"
+      },
+      "outputs": [],
+      "source": [
+        "def extract_rag_id(rag_resource_name):\n",
+        "    \"\"\"Extracts the RAG ID from a RAG resource name.\n",
+        "\n",
+        "    Args:\n",
+        "      rag_resource_name: The full name of the RAG resource.\n",
+        "\n",
+        "    Returns:\n",
+        "      The RAG ID extracted from the resource name.\n",
+        "    \"\"\"\n",
+        "    return rag_resource_name.split(\"/\")[-1]\n",
+        "\n",
+        "\n",
+        "def extract_doc_id(file_path):\n",
+        "    \"\"\"Extracts the document ID from a file path.\n",
+        "\n",
+        "    The function assumes the document ID is the number before the '.txt' extension in the file path.\n",
+        "\n",
+        "    Args:\n",
+        "      file_path: The path to the file.\n",
+        "\n",
+        "    Returns:\n",
+        "      The document ID extracted from the file path, or None if no ID is found.\n",
+        "    \"\"\"\n",
+        "    # Use regular expression to find the number before .txt, with optional /\n",
+        "    match = re.search(r\"/?(\\d+)\\.txt$\", file_path)\n",
```

Comment:
Aren't there methods/properties in the `rag` module already for this?

Reply:
Removed `extract_rag_id`.

`def extract_doc_id(file_path)` is a function specifically used for evaluation purposes with BEIR dataset. I don't think the `rag` module would have this.

Reply:
Ok, sounds good.